src/rpc/guards/HandshakeCompletedGuard.ts

```diff
@@ -0,0 +1,50 @@
+import Rpc from "@/rpc/Rpc";
+import ATransport from "@/transport/ATransport";
+import { AGuard } from "@/rpc/guards/AGuard";
+import ARpcService from "@/rpc/ARpcService";
+import ARpcMethods from "@/rpc/ARpcMethods";
+
+export interface HandshakeCompletedGuardOptions {
+    onFailure?: (rpc: Rpc, transport: ATransport) => void;
+}
+
+export class HandshakeCompletedGuard extends AGuard<ARpcService<ARpcMethods>> {
```

Comment:
Any plans to add more guards?
how many more?
an abstract class just for a single guard?

Reply:
I'm adding isSynced to some endpoints when I submit the spectate test PR + I want to make RPC services extendable, so the consumer can add their services and these guards have a simple interface if they require protection on their endpoints

Later we might have different algos for rate limiting dependent on the RPC endpoint - a critical one like `stateTransitions` will be allocated higher bandwidth/priority while a non-critical one like "messages" can be cut under load